cmd/meroxa/root/apps/deploy.go

```diff
@@ -524,6 +525,68 @@ func (d *Deploy) validate(ctx context.Context) error {
 	return turbineCLI.ValidateBranch(ctx, d.logger, d.path)
 }
 
+func (d *Deploy) getResourceCheckErrorMessage(ctx context.Context, resourceNames []string) string {
+	var errStr string
+	for _, name := range resourceNames {
+		resource, err := d.client.GetResourceByNameOrID(ctx, name)
+		if err != nil {
+			if errStr != "" {
+				errStr += "; "
+			}
+
+			if strings.Contains(err.Error(), "could not find") {
+				errStr += fmt.Sprintf("could not find resource %q", name)
+			} else {
+				errStr += err.Error()
+			}
+		} else if resource.Status.State != meroxa.ResourceStateReady {
+			if errStr != "" {
+				errStr += "; "
+			}
+			errStr += fmt.Sprintf("resource %q is not ready and usable", resource.Name)
+		}
+	}
+	return errStr
+}
+
+func (d *Deploy) checkResourceAvailability(ctx context.Context) error {
+	resourceCheckMessage := fmt.Sprintf("Checking resource availability for application %q (%s) before deployment...", d.appName, d.lang)
+
+	d.logger.StartSpinner("\t", resourceCheckMessage)
+
+	var resourceNames []string
+	var err error
+
+	switch d.lang {
+	case GoLang:
+		resourceNames, err = turbineGo.GetResourceNames(ctx, d.logger, d.path, d.appName)
+	case JavaScript:
+		resourceNames, err = turbineJS.GetResourceNames(ctx, d.logger, d.path, d.appName)
+	case Python:
+		resourceNames, err = turbinePY.GetResourceNames(ctx, d.logger, d.path, d.appName)
+	}
+
+	if err != nil {
+		return fmt.Errorf("unable to read resource definition from app: %s", err.Error())
+	}
+
+	if len(resourceNames) == 0 {
+		return errors.New("no resources defined in your Turbine app")
+	}
+
+	errStr := d.getResourceCheckErrorMessage(ctx, resourceNames)
+
+	if errStr != "" {
+		errStr += ";\n\n ⚠️  Run 'meroxa resource list' to verify that the names of resources " +
+			"defined in your Turbine app are identical to the resources you have created on the Meroxa platform before deploying again"
```

Comment:
Nitpicking (again). When mentioning Meroxa products, I tend to use the uppercase form (i.e.: `Platform` instead of `platform`.

```suggestion
			"defined in your Turbine app are identical to the resources you have created on the Meroxa Platform before deploying again"
```

Reply:
Thank you for the suggestion!